### PR TITLE
Made "Medication, Order" cumulative medication duration calculate correctly.

### PR DESCRIPTION
**Pull request by edeyoung**
This addresses https://oncprojectracking.healthit.gov/support/browse/BONNIE-172.

Usually, cumulative medication duration is calculated based off of the fulfillment history, dose, and regimen. However, for "Medication, Order", the medication has not yet been filled, so fulfillment history and dose don't make sense. Maybe more importantly, these fields are not available in the QRDA for the "Medication, Order" template.

Instead, cumulative medication duration should be calculated based on the maximum allowed administrations (`repeatNumber` in the QRDA) and the regimen (`effectiveTime` with `period` subelement in the QRDA).

Additionally, fulfillment information should not be allowed to be entered for "Medication, Order".

Specific changes for this repository include changing the cumulative medication duration calculation so that it accounts for "Medication, Order".

This pull request depends on the `medication_order_cmd` branches in **patient-api, bonnie, and health-data-standards.**

### CODE REVIEW

File: app/assets/javascripts/patient_api_extension.js.coffee [hunk 1]
```diff
@@ -115,7 +115,11 @@ hQuery.Medication::cumulativeMedicationDuration = (dateRange) ->
   #doses not total amount. Will need to flush this out more at a later point in time.
   #Considering that liquid meds are probaly dispensed as total volume ex 325ml with a dose of
   #say 25ml per dose.  Will definatley need to revisit this.
-  this.fulfillmentTotals(dateRange) if this.administrationTiming() && this.dose() && @json['fulfillmentHistory']
+  if this.administrationTiming() && this.dose() && @json['fulfillmentHistory']
+    this.fulfillmentTotals(dateRange)
+  else if this.administrationTiming() && this.allowedAdministrations()
+    # this happens if we have a Medication, Order.
+    this.allowedAdministrations() / this.administrationTiming().dosesPerDay()
```
Comment (thecristen) [inline app/assets/javascripts/patient_api_extension.js.coffee]: What if `this.administrationTiming().dosesPerDay()` returns 0?

Comment (thecristen) [inline app/assets/javascripts/patient_api_extension.js.coffee]: It looks like `this.allowedAdministrations() / this.administrationTiming().dosesPerDay()` will return a length of time described in number of days. Is this consistent with what `this.fulfillmentTotals(dateRange)` returns?

Comment (thecristen) [inline app/assets/javascripts/patient_api_extension.js.coffee]: I notice that `this.allowedAdministrations() / this.administrationTiming().dosesPerDay()` does not depend on the `dateRange` argument... is there ever a scenario where
`this.administrationTiming() && this.dose() && @json['fulfillmentHistory']` is `true` but there's no available `dateRange` (in which case this new bit of code offers the opportunity to use the `allowedAdministrations`-based calculated as a fallback)?

Comment (edeyoung) [inline app/assets/javascripts/patient_api_extension.js.coffee]: yes

**Pull request merged**